### PR TITLE
Add C implementation for `@stdlib/math/base/special/asin`

### LINKED ISSUE #648: [RFC]: Add C implementation for `@stdlib/math/base/special/asin`
### Description

This RFC proposes adding a C implementation for [`@stdlib/math/base/special/asin`](https://github.com/stdlib-js/stdlib/blob/b43a75923e47775446d3b6d7219ee42e07f7ce63/lib/node_modules/%40stdlib/math/base/special/asin).

```c
double stdlib_base_asin( const double x );
```

### Related Issues

None.

### Questions

No.

### Other

No.

### Checklist

- [X] I have read and understood the [Code of Conduct](https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md).
- [X] Searched for existing issues and pull requests.
- [X] The issue name begins with `RFC:`.

### PR DESCRIPTION
**Pull request by Pranavchiku**
Resolves #648.

## Checklist

- [x] update readme.md
- [x] include.gypi
- [x] binding.gyp
- [x] include/stdlib/math/base/special/
- [x] evalpoly.js
- [x] src
- [x] manifest.json
- [x] lib
- [x] examples
- [x] benchmark
- [x] test
* * *

@kgryte

### CODE REVIEW

File: lib/node_modules/@stdlib/math/base/special/asin/scripts/evalrational.js
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/asin/scripts/evalrational.js]: You need `evalrational-compile-c` here which is on the latest `develop`.

Comment (Pranavchiku) [reply lib/node_modules/@stdlib/math/base/special/asin/scripts/evalrational.js]: Thanks! I was gonna ask to check it :)

**Comment by Pranavchiku** [comment]
All tests and benchmarks cleared, this PR can have a final review.

**Comment by Pranavchiku** [comment]
@kgryte when `evalrational.js` generates c-functions it defines them as 
```c
static double rational_rs( const double x )
```
and then the const variable `x` is used in the assignment, which throws an error while installing node addons.

File: lib/node_modules/@stdlib/math/base/special/asin/src/main.c
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/special/asin/src/main.c:114]: Refer previous comment

**Comment by kgryte** [comment]
@Pranavchiku Thanks for finding this bug. I'll fix and push a change to `develop` shortly.

**Comment by kgryte** [comment]
@Pranavchiku This should now be fixed on the latest `develop`.

File: lib/node_modules/@stdlib/math/base/special/asin/README.md
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/asin/README.md]:
```suggestion
Computes the [arcsine][arcsine] of a number (in radians).
```

File: lib/node_modules/@stdlib/math/base/special/asin/README.md
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/asin/README.md:112]:
```suggestion

out = stdlib_base_asin( -3.141592653589793/6.0 );
```

File: lib/node_modules/@stdlib/math/base/special/asin/README.md
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/asin/README.md:147]:
```suggestion

int main() {
```

File: lib/node_modules/@stdlib/math/base/special/asin/README.md
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/asin/README.md]:
```suggestion
    double x[] = { -1.0, -0.78, -0.56, -0.33, -0.11, 0.11, 0.33, 0.56, 0.78, 1.0 };
```

File: lib/node_modules/@stdlib/math/base/special/asin/benchmark/c/native/benchmark.c
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/asin/benchmark/c/native/benchmark.c]:
```suggestion
		printf( "# c::native::%s\n", NAME );
```

File: lib/node_modules/@stdlib/math/base/special/asin/examples/c/example.c
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/asin/examples/c/example.c]:
```suggestion
    double x[] = { -1.0, -0.78, -0.56, -0.33, -0.11, 0.11, 0.33, 0.56, 0.78, 1.0 };
```

File: lib/node_modules/@stdlib/math/base/special/asin/include/stdlib/math/base/special/asin.h
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/asin/include/stdlib/math/base/special/asin.h]:
```suggestion
* Computes the arcsine of a number.
```

File: lib/node_modules/@stdlib/math/base/special/asin/lib/native.js
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/asin/lib/native.js:32]:
```suggestion
* @private
* @param {number} x - input value
```

File: lib/node_modules/@stdlib/math/base/special/asin/src/main.c
Comment (kgryte) [reply lib/node_modules/@stdlib/math/base/special/asin/src/main.c:114]: After merging, I'll rerun the script to include the latest generated code.

File: lib/node_modules/@stdlib/math/base/special/asin/src/main.c
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/asin/src/main.c]:
```suggestion
* Computes the arcsine of a number.
```

**Review by kgryte — APPROVED** [review]

**Pull request merged**
